tests/hypothesis_strategies.py

```diff
@@ -0,0 +1,158 @@
+"""Hypothesis strategies."""
+import random
+
+import hypothesis.strategies as st
+import numpy as np
+from hypothesis import assume
+from hypothesis.extra.pandas import columns, data_frames
+
+from baybe.exceptions import NumericalUnderflowError
+from baybe.parameters.categorical import (
+    CategoricalEncoding,
+    CategoricalParameter,
+    TaskParameter,
+)
+from baybe.parameters.custom import CustomDiscreteParameter
+from baybe.parameters.numerical import (
+    NumericalContinuousParameter,
+    NumericalDiscreteParameter,
+)
+from baybe.parameters.substance import SubstanceEncoding, SubstanceParameter
+from baybe.utils import DTypeFloatNumpy
+
+_largest_lower_interval = np.nextafter(
+    np.nextafter(np.inf, 0, dtype=DTypeFloatNumpy), 0, dtype=DTypeFloatNumpy
+)
+"""
+The largest possible value for the lower end of a continuous interval such that there
+still exists a larger but finite number for the upper interval end.
+"""
+
+_decorrelate = st.one_of(
+    st.booleans(),
+    st.floats(min_value=0.0, max_value=1.0, exclude_min=True, exclude_max=True),
+)
+"""A strategy that generates decorrelation settings."""
+
+parameter_name = st.text(min_size=1)
+"""A strategy that generates parameter names."""
+
+categories = st.lists(st.text(min_size=1), min_size=2, unique=True)
+"""A strategy that generates parameter categories."""
+
+
+@st.composite
+def smiles(draw: st.DrawFn):
+    """Generate short SMILES strings."""
+    n_atoms = draw(st.integers(min_value=0, max_value=19))
+    string = "C"
+    for _ in range(n_atoms):
+        next_atom = random.choice("CNO") if string[-1] == "C" else random.choice("C")
+        string += next_atom
+    return string
+
+
+@st.composite
+def substance_data(draw: st.DrawFn):
+    """Generate data for class:`baybe.parameters.substance.SubstanceParameter`."""
+    names = draw(st.lists(st.text(min_size=1), min_size=2, max_size=10, unique=True))
+    substances = draw(st.lists(smiles(), min_size=len(names), max_size=len(names)))
+    return dict(zip(names, substances))
+
+
+@st.composite
+def custom_encodings(draw: st.DrawFn):
+    """Generate data for class:`baybe.parameters.custom.CustomDiscreteParameter`."""
```

Comment:
I think there is a : missing in front of class. Please check and verify this (and all other similar) link(s).

Reply:
Absolutely. Damn copy-paste 😄 thanks! Got unnoticed because we are not building the doc for our tests. Could debate if we actually want to do it or not.

Reply:
Fixed it, btw

Reply:
Is there any reason not to include building the documentation as a test? We do not have broken links towards github pages, and iirc, these were the only ones preventing us from including this, right? If you feel that we might want to discuss this, please open a corresponding discussion (and feel free to resolve here in any case)

Reply:
I didn't mean "building the documentation as a test" but "building the documentation for our test code", which is currently not done since the test code is also not displayed on our github.io page. I was just saying: the error would have been noticed if the test folder was part of the doc generation.... Since it is not, it's also a bit questionable why I've put the class-links there in the first place. So what should we do? 
1) Exclude the test folder from the pipeline and effectively have no check for links / remove the links in the first place 
2) Generate the docs to check the links but not put in on the web
3) Generate the docs to check the links and put it also on the web

Reply:
Moved this to a [separate discussion](https://github.com/emdgroup/baybe/discussions/63) since not related to the PR. Will resolve